cpp/src/arrow/python/serialize.cc

```diff
@@ -63,297 +70,195 @@ class SequenceBuilder {
       : pool_(pool),
         types_(::arrow::int8(), pool),
         offsets_(::arrow::int32(), pool),
-        nones_(pool),
-        bools_(::arrow::boolean(), pool),
-        ints_(::arrow::int64(), pool),
-        py2_ints_(::arrow::int64(), pool),
-        bytes_(::arrow::binary(), pool),
-        strings_(pool),
-        half_floats_(::arrow::float16(), pool),
-        floats_(::arrow::float32(), pool),
-        doubles_(::arrow::float64(), pool),
-        date64s_(::arrow::date64(), pool),
-        tensor_indices_(::arrow::int32(), pool),
-        ndarray_indices_(::arrow::int32(), pool),
-        buffer_indices_(::arrow::int32(), pool),
-        list_offsets_({0}),
-        tuple_offsets_({0}),
-        dict_offsets_({0}),
-        set_offsets_({0}) {}
+        type_map_(PythonType::MAX) {
```

Comment:
Perhaps fill it with an initialization value? `-1` for example?

Reply:
This is the number of types (now NUM_PYTHON_TYPES), to have the right size here.

Reply:
Yes, I meant `type_map_(NUM_PYTHON_TYPES, -1)`.

Reply:
got it! I changed it!